sqlglot/expressions.py

```diff
@@ -238,6 +238,9 @@ def type(self, dtype: t.Optional[DataType | DataType.Type | str]) -> None:
             dtype = DataType.build(dtype)
         self._type = dtype  # type: ignore
 
+    def is_type(self, *dtypes) -> bool:
+        return self.type is not None and self.type.is_type(*dtypes)
```

Comment:
self.type and self.type.is_type(*dtypes)

Reply:
mypy doesn't like that because the return type is `t.Optional[bool]`. Quite annoying. Do you know a way around that?

Reply:
ah…